aws/table_aws_cloudwatch_metric_statistic_data_point.go

```diff
@@ -0,0 +1,173 @@
+package aws
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+	cloudwatchv1 "github.com/aws/aws-sdk-go/service/cloudwatch"
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsCloudWatchMetricStatisticDataPoint(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_cloudwatch_metric_statistic_data_point",
+		Description: "AWS CloudWatch Metric Statistic Data Point",
+		List: &plugin.ListConfig{
+			Hydrate: listCloudWatchMetricStatisticDataPoints,
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"InvalidParameterValue"}),
+			},
+			KeyColumns: []*plugin.KeyColumn{
+				{
+					Name:    "metric_name",
+					Require: plugin.Required,
+				},
+				{
+					Name:    "namespace",
+					Require: plugin.Required,
+				},
+				{
+					Name:      "timestamp",
+					Operators: []string{">", ">=", "=", "<", "<="},
+					Require:   plugin.Required,
+				},
+				{
+					Name:    "period",
+					Require: plugin.Optional,
+				},
+				{
+					Name:    "dimensions",
+					Require: plugin.Optional,
+				},
+			},
+		},
+		GetMatrixItemFunc: SupportedRegionMatrix(cloudwatchv1.EndpointsID),
+		Columns: awsRegionalColumns(cwMetricColumns([]*plugin.Column{
+			{
+				Name:        "label",
+				Description: "A label for the specified metric.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "period",
+				Description: "The granularity, in seconds, of the returned data points.",
+				Type:        proto.ColumnType_INT,
+			},
+			{
+				Name:        "dimensions",
+				Description: "The dimensions for the metric.",
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Label"),
+			},
+		})),
+	}
+}
+
+type MetricStatistics struct {
+	MetricName *string
+	Namespace  *string
+	Period     *int32
+	Label      *string
+	Dimensions []types.Dimension
+	types.Datapoint
+}
+
+//// LIST FUNCTION
+
+func listCloudWatchMetricStatisticDataPoints(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	// set the input parameters
+	params := &cloudwatch.GetMetricStatisticsInput{
+		MetricName: aws.String(d.EqualsQuals["metric_name"].GetStringValue()),
+		Namespace:  aws.String(d.EqualsQuals["namespace"].GetStringValue()),
+		Statistics: types.Statistic.Values(types.StatisticMaximum),
```

Comment:
What value is in `types.StatisticMaximum`? Are we getting all statistics by passing in this input value?

Reply:
`types.StatisticMaximum` holds the value `maximum`
If we pass any types.Statistic to the `types.Statistic.Values` function, in return you will get all.
```
// Values returns all known values for Statistic. Note that this can be expanded in
// the future, and so it is only as up to date as the client. The ordering of this
// slice is not guaranteed to be stable across updates.
func (Statistic) Values() []Statistic {
	return []Statistic{
		"SampleCount",
		"Average",
		"Sum",
		"Minimum",
		"Maximum",
	}
}
```